pydatastructs/miscellaneous_data_structures/Fibonacci_Heap.py

```diff
@@ -0,0 +1,189 @@
+import math
+
+
+class FibonacciHeap:
+
+    # internal node class
+    class Node:
+        def __init__(self, key, value):
+            self.key = key
+            self.value = value
+            self.parent = self.child = self.left = self.right = None
+            self.degree = 0
+            self.mark = False
+
+    # function to iterate through a doubly linked list
```

Comment:
For documentation please follow the numpy doc string guide. See other modules for the pattern of how we write doc strings.